src/com/rabbitmq/client/Connection.java

```diff
@@ -94,6 +94,13 @@
     Map<String, Object> getClientProperties();
 
     /**
+     * Get connection name client property value
+     *
+     * @return string connection name from client properties, or null if there is not such property.
+     */
+    String getConnectionName();
```

Comment:
It should be `getClientProvidedConnectionName` or `getCustomConnectionName`. I will do the renaming.